proxy/src/main/java/com/velocitypowered/proxy/connection/client/LoginSessionHandler.java

```diff
@@ -127,16 +128,25 @@ public boolean handle(EncryptionResponse packet) {
           } else if (profileResponse.getStatusCode() == 204) {
             // Apparently an offline-mode user logged onto this online-mode proxy.
             inbound.disconnect(VelocityMessages.ONLINE_MODE_ONLY);
+            server.getEventManager().fireAndForget(new LoginFailEvent(inbound,
```

Comment:
Consider a `disconnect()` private function for cases like these

Reply:
Done in the new commit. Thanks for the tip.